forum/qa-content/async-questions-list.js

```diff
@@ -1,21 +1,143 @@
-window.addEventListener('DOMContentLoaded', function() {
+document.addEventListener('DOMContentLoaded', function runAsyncQuestionsList() {
+	const { pathname, hostname } = window.location;
+	const isMainOrActivityPage = /^$|\/$|^(\/?)activity/.test(pathname);
 
-	// Is this the main page?
-	if ($('.qa-template-qa').length === 0) {
-		return // no
+	if (!isMainOrActivityPage) {
+		return;
 	}
 
-	// connect to websocket
-	var socket = new WebSocket('ws://localhost:3001')
+	const QA_Q_LIST_SELECTOR = '.qa-q-list';
+	const questionList = document.querySelector(QA_Q_LIST_SELECTOR);
+	const newContentNotifier = createNewContentNotifier();
+	const webSocket = setupWebSocketConnection();
 
-	// get question list node
-	var $questionList = $('.qa-q-list')
+	function setupWebSocketConnection() {
+		const webSocket = new WebSocket(`ws://${hostname}:3000`);
+		webSocket.addEventListener('open', onOpen);
+		webSocket.addEventListener('message', onMessage);
+		webSocket.addEventListener('close', onClose);
+		webSocket.addEventListener('error', onError);
 
-	// listen to message
-	socket.addEventListener('message', function(event) {
+		window.addEventListener('beforeunload', () => {
+			webSocket.close();
+			newContentNotifier.disconnectIntersectionObserver();
+		});
 
-		var data = window.JSON.parse(event.data)
-		$questionList.html(data.html)
+		return webSocket;
+	}
+
+	function onOpen(event) {
+		webSocket.send(JSON.stringify({ pathname }));
+	}
+
+	function onMessage(event) {
+		// TODO: data variable will be used when WebSocket will eventually send HTML content
```

Comment:
komentarz todo - czyli niegotowe?

Reply:
To jest póki co wersja, która ma sprawdzić, czy WebSocket po poprawkach i aktualizacji bibliotek działa stabilnie na produkcji. Dalsze ficzery, w tym przesyłanie danych (odbiór jest właśnie zakomentowany jako TODO w powyższym fragmencie kodu) będą dodawane w kolejnych etapach.